tools/myst/enc/enc.c

```diff
@@ -513,11 +513,10 @@ static long _enter(void* arg_)
         report_native_tids = options->report_native_tids;
         max_affinity_cpus = options->max_affinity_cpus;
 
-        if (strlen(options->rootfs) >= PATH_MAX)
-        {
-            fprintf(stderr, "rootfs path too long (> %u)\n", PATH_MAX);
-            goto done;
-        }
+        /* rootfs buffer content set by the host side. Max length of the string
+         * is PATH_MAX-1. Enforce NULL terminator at the end of the buffer.
+         */
+        options->rootfs[PATH_MAX - 1] = '\0';
```

Comment:
It would be nice to retain the error condition above if possible rather than silently truncating the buffer (in the case where the buffer has no null terminators). Maybe this:

```
        if (strnlen(options->rootfs, PATH_MAX) == PATH_MAX)
```